### PR TITLE
feat: Propagate zbus errors upwards to prepare for handling

### PR DESCRIPTION
**Pull request by flukejones**
- Propagate most of the zbus errors upwards to callers, this does affect pub fn in some places.
- Bump minor version

This doesn't fully handle errors yet as the dbus errors need handling also. Regarding dbus, would it be best to drop the dependency? Zbus currently does everything dbus-rs does and does not require an external library with all the cruft that comes with FFI.

### CODE REVIEW

**Comment by zeenix** [comment]
> Regarding dbus, would it be best to drop the dependency? Zbus currently does everything dbus-rs does and does not require an external library with all the cruft that comes with FFI.

I said the same to @hoodie in a PM yesterday. :) There actually was a reason until recently: zbus required much newer MSRV so it was problematic for users that needed to ensure their projects can be easily packaged for distros.

However, now zbus' MSRV (1.60) is just one higher than that of dbus-rs (1.59) and AFAIK there is no distro that ships 1.59. So I think it's time to cut dbus-rs loose and simplify the implementation and usage (so users don't have to choose between features), apart from the cross-compiling (and maybe other) benefits.

**Comment by zeenix** [comment]
@flukejones btw, I believe @hoodie asked you to split your commit? [Atomic commits](https://www.aleksandrhovhannisyan.com/blog/atomic-git-commits/) is a practice all good projects that value quality adhere to (zbus includded) so I'll recommend getting into that practice.

**Comment by flukejones** [comment]
@zeenix yes, this PR is only error handling, nothing else.

**Comment by zeenix** [comment]
> Bump minor version

@flukejones Ah that's bumping this crates version, gotcha.

File: src/lib.rs [hunk 1]
```diff
@@ -162,23 +165,26 @@ mod hints;
 mod miniver;
 mod notification;
 mod timeout;
+pub(crate) mod urgency;
```
Comment (hoodie) [inline src/lib.rs:168]: please leave this at the bottom with the other `pub mod`s

**Review by hoodie — CHANGES_REQUESTED** [review]

File: Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "notify-rust"
-version = "4.5.10"
+version = "4.6.0"
```
[the remark this answers is not in the document]
Comment (flukejones) [reply Cargo.toml:3]: Breaking changes to API. Wasn't actually sure how to handle here. (I'll change)

File: src/error.rs [hunk 3]
```diff
@@ -28,6 +28,9 @@ pub enum ErrorKind {
     #[cfg(all(feature = "zbus", unix, not(target_os = "macos")))]
     Zbus(zbus::Error),
 
+    #[cfg(all(feature = "zbus", unix, not(target_os = "macos")))]
+    ZbusFdo(zbus::fdo::Error),
```
[the remark this answers is not in the document]
Comment (flukejones) [reply src/error.rs:32]: rust-analyser said so?

File: src/lib.rs (hunk 1 above)
Comment (flukejones) [reply src/lib.rs:168]: Added by rust-ananlyser. Might need a rustfmt.toml

File: Cargo.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (hoodie) [reply Cargo.toml:3]: this repo uses conventional commit messages, so if you have a `feat:` at the beginning of the message the next release will have a bumped minor version automatically

File: src/error.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (zeenix) [reply src/error.rs:32]: What exactly did RA say? :) `From<zbus::fdo::Error> for zbus::Error` is implemented so you probably just need a `.map_err(Into::into)` where it complains?

[the remark this answers is not in the document]
Comment (flukejones) [reply src/error.rs:32]: I don't recall exactly. I'll make the requested changes when I get some time.

**Comment by zeenix** [comment]
@flukejones Hey. any chance you can finish this soon?

**Comment by flukejones** [comment]
Closing as I don't think I'll get around to finishing this.

**Pull request closed without merging**